src/test/java/integration/EventsTest.java

```diff
@@ -882,8 +884,9 @@ public void given_multibranchWithSourcesAndTwoBranches_when_indexing_then_deadBr
             prj.scheduleBuild2(0).getFuture().get();
             r.waitUntilNoActivity();
             assertThat("Indexing applies dead branch cleanup",
-                    prj.getItems(), containsInAnyOrder(master));
```

Comment:
http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#containsInAnyOrder(T...)

> Creates an order agnostic matcher for Iterables that matches when a single pass over the examined Iterable yields a series of items, each logically equal to one item anywhere in the specified items. For a positive match, the examined iterable must be of the same length as the number of specified items.
> N.B. each of the specified items will only be used once during a given examination, so be careful when specifying items that may be equal to more than one entry in an examined iterable.


So try 

`assertThat(Arrays.asList("foo", "bar"), containsInAnyOrder("bar")))` 

and watch it fail

`assertThat(Arrays.asList("foo", "bar"), containsInAnyOrder("bar","foo")))` 

Will pass, while

`assertThat(Arrays.asList("foo", "bar"), containsInAnyOrder("manchu", "bar", "foo")))` 

will fail.

So you were over-complicating the reported failure in the event of a mismatch.